packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts

```diff
@@ -624,9 +624,12 @@ export class BaseTypesVisitor<
   }
 
   getInputObjectOneOfDeclarationBlock(node: InputObjectTypeDefinitionNode): DeclarationBlock {
+    // Since oneOf declarations always result in a union if having
+    // more than 1 field, we have to force a declaration kind of `type`
+    const declarationKind = node.fields.length === 1 ? this._parsedConfig.declarationKind.input : 'type';
```

Comment:
I'm a bit unsure about making a difference between single and multiple fields.
Please let me know what you would prefer.